components/chainhook-sdk/src/observer/mod.rs

```diff
@@ -363,6 +363,35 @@ impl ChainhookStore {
     }
 }
 
+#[derive(Debug, Default, Serialize, Clone)]
+pub struct ReorgMetrics {
+    timestamp: i64,
+    applied_blocks: usize,
+    rolled_back_blocks: usize,
+}
+
+#[derive(Debug, Default, Serialize, Clone)]
+pub struct ChainMetrics {
+    pub tip_height: u64,
+    pub last_reorg: Option<ReorgMetrics>,
+    pub last_block_ingestion_at: u128,
+    pub registered_predicates: usize,
+    pub deregistered_predicates: usize,
+}
+
+impl ChainMetrics {
+    pub fn deregister_prediate(&mut self) {
+        self.registered_predicates -= 1;
+        self.deregistered_predicates += 1;
+    }
+}
+
+#[derive(Debug, Default, Serialize, Clone)]
+pub struct ObserverMetrics {
+    pub bitcoin: ChainMetrics,
+    pub stacks: ChainMetrics, // todo: do users sometimes start with just bitcoin, making stacks unnecessary?
```

Comment:
@lgalabru

I can remove the todo here, I just wanted to confirm that we're okay with `stacks` being a permanent member of these metrics. It looks like for most of this file we're assuming that stacks is enabled, but maybe I'm missing something and should make the stacks metrics optional.

Reply:
Hmm, I wondering if other stuff would have to change for that to make sense. It seems the event observer always assumes a stacks chain is running. All of the configuration has stacks data as non-optional, and we're always exposing stacks routes:
```
let mut routes = rocket::routes![
        http::handle_ping,
        http::handle_new_bitcoin_block,
        http::handle_new_stacks_block,
        http::handle_new_microblocks,
        http::handle_new_mempool_tx,
        http::handle_drop_mempool_tx,
        http::handle_new_attachement,
        http::handle_mined_block,
        http::handle_mined_microblock,
    ];
```

Is that a correct assessment? If so, should I create an issue to remove logic from the event observer that always enables stacks event observation?

Reply:
My apologies for the confusion. If the event observer is started, then we know for sure that Stacks + Bitcoin are observed - we don't (or should not) start the server if the bitcoin block signaling is happening through Zeromq.

Reply:
Okay great, thanks. I've removed the TODO